src/Core/x/xGroup.cpp

```diff
@@ -1,39 +1,177 @@
 #include "xGroup.h"
 
+#include "../Core/x/xMemMgr.h"
+#include "zScene.h"
+
 #include <types.h>
 
-// func_8002CAA0
-#pragma GLOBAL_ASM("asm/Core/x/xGroup.s", "xGroupInit__FPvPv")
+void xGroupInit(void* b, void* asset)
+{
+    xGroupInit((xBase*)b, (xGroupAsset*)asset);
+}
+
+#if 1
 
 // func_8002CAC0
 #pragma GLOBAL_ASM("asm/Core/x/xGroup.s", "xGroupInit__FP5xBaseP11xGroupAsset")
 
-// func_8002CB70
-#pragma GLOBAL_ASM("asm/Core/x/xGroup.s", "xGroupSetup__FP6xGroup")
+#else
+
+// There is an instruction swap that seems to be throwing everything off.
+void xGroupInit(xBase* b, xGroupAsset* asset)
+{
+    xBaseInit(b, (xBaseAsset*)asset);
+    b->eventFunc = xGroupEventCB;
+    ((xGroup*)b)->asset = asset;
+    if (b->linkCount)
+    {
+        b->link =
+            (xLinkAsset*)(&asset->link + ((xGroup*)b)->asset->itemCount); // lwz and lhz swap here.
```

Comment:
```
(xLinkAsset*)(&asset->link
```

will need to be changed to

```
(xLinkAsset*)(asset + 1
```

in regards to my other comment.

Reply:
My bad, you'll probably need to cast `asset + 1` to a uint32* actually so it stays correct: 
```
(xLinkAsset*)((uint32*)(asset + 1) + ((xGroup*)b)->asset->itemCount)
```